io/src/read_bfield.cpp

```diff
@@ -0,0 +1,66 @@
+/** TRACCC library, part of the ACTS project (R&D line)
+ *
+ * (c) 2025 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Local include(s).
+#include "traccc/io/read_bfield.hpp"
+
+#include "traccc/io/utils.hpp"
+
+// System include(s).
+#include <format>
+#include <fstream>
+#include <stdexcept>
+
+namespace traccc::io {
+namespace binary {
+
+void read_bfield(covfie::field<inhom_bfield_backend_t<traccc::scalar>>& field,
+                 std::string_view filename,
+                 std::unique_ptr<const Logger> ilogger) {
+
+    // Set up a local logger.
+    TRACCC_LOCAL_LOGGER(std::move(ilogger));
+
+    // Open the file.
+    std::ifstream ifile{get_absolute_path(filename),
+                        std::ios::binary | std::ios::in};
+    if (!ifile.is_open()) {
+        throw std::invalid_argument(
+            std::format("Failed to open magnetic field file: {}", filename));
+    }
+
+    // Check the magic header, to see that it's really a "covfie file".
+    if (auto hdr = covfie::utility::read_binary<std::uint32_t>(ifile);
+        hdr != covfie::utility::MAGIC_HEADER) {
+        throw std::invalid_argument(std::format(
+            "Input file ({}) is not a valid covfie file", filename));
+    }
+
+    // Reset the read position to the beginning of the file.
+    ifile.seekg(0);
```

Comment:
This check is also done by the IO code in covfie, so it's somewhat redundant here.